lib/soapy_cake/request.rb

```diff
@@ -38,6 +38,10 @@ def to_s
       "#{role}:#{service}:#{method}:#{version} #{opts.to_json}"
     end
 
+    def read_only?
+      (API_CONFIG.dig('read_only', role, service) || []).include?(method)
```

Comment:
It sure beats `try(:[], 'read_only').try(:[], role).try(:[], service)`, anyway.  Hopefully won't get us into too much trouble :)